src/NodeServices/PythonServices.cs

```diff
@@ -125,8 +124,10 @@ internal static readonly Lazy<PythonEngineManager>
         [Obsolete("AvailableEngines field will be replaced in a future Dynamo release.")]
         public ObservableCollection<PythonEngine> AvailableEngines;
 
+        internal Dictionary<string, Tuple<string, string>> PythonEnginePackageMap;
```

Comment:
why not add the Package and Version properties to the PythonEngine class ?
All python engines are loaded from `DSCPython` (part of Dynamo) or from the `PackageManager.PackageLoader.TryLoadPackageIntoLibrary`. You could easily send the pkg information through the `LoadPythonEngine` arguments

Reply:
Yes, but it is not straight-forward, the python engine is loaded as an extension, and extensions are loaded after packages, so we do not know if an engine was added after the packages are loaded.